public/pages/Findings/containers/Findings/Findings.tsx

```diff
@@ -257,33 +404,58 @@ class Findings extends Component<FindingsProps, FindingsState> {
   };
 
   generateVisualizationSpec() {
-    const visData: FindingVisualizationData[] = [];
-
-    this.state.filteredFindings.forEach((finding: FindingItemType) => {
-      const findingTime = new Date(finding.timestamp);
-      findingTime.setMilliseconds(0);
-      findingTime.setSeconds(0);
-      finding.detectionType === 'Threat intelligence';
-      const ruleLevel =
-        finding.detectionType === 'Threat intelligence'
-          ? 'high'
-          : this.state.rules[finding.queries[0].id].level;
-      visData.push({
-        finding: 1,
-        time: findingTime.getTime(),
-        logType: finding.detector._source.detector_type,
-        ruleSeverity:
-          ruleLevel === 'critical' ? ruleLevel : (finding as any)['ruleSeverity'] || ruleLevel,
+    const visData: (FindingVisualizationData | ThreatIntelFindingVisualizationData)[] = [];
+    const { selectedTabId, findingStateByTabId } = this.state;
+
+    const findingsState =
+      selectedTabId === FindingTabId.DetectionRules
+        ? findingStateByTabId[FindingTabId.DetectionRules]
+        : findingStateByTabId[FindingTabId.ThreatIntel];
+    const groupBy = findingsState.groupBy;
+    let specGetter;
+
+    if (selectedTabId === FindingTabId.DetectionRules) {
+      (findingsState.filteredFindings as FindingItemType[]).forEach((finding: FindingItemType) => {
+        const findingTime = new Date(finding.timestamp);
+        findingTime.setMilliseconds(0);
+        findingTime.setSeconds(0);
+        finding.detectionType === 'Threat intelligence';
+        const ruleLevel =
+          finding.detectionType === 'Threat intelligence'
+            ? 'high'
+            : (findingsState as DetectionRulesFindingsState).rules[finding.queries[0].id].level;
```

Comment:
```
 : (findingsState as DetectionRulesFindingsState).rules[finding.queries[0]?.id]?.level || undefined?
```

Reply:
This is existing logic, just moved it inside the tab state